### PR TITLE
Port gosec G401

### PR DESCRIPTION
**Pull request by ulziibay**
As titled. Adds 4 rules to warn against the usage of sha1, md5, rc4, and DES

## Testing
Locally run
`semgrep -f go/gosec/bad_crypto go/gosec`

```
/Users/ulziibayarotgonbaatar/Workspace/sgrep-rules/go/gosec/bad_crypto/use_of_weak_crypto_des.go
rule:go.gosec.bad_crypto.use_of_DES: Use of weak cryptographic primitive DES
16:	_, err := des.NewTripleDESCipher(tripleDESKey)

/Users/ulziibayarotgonbaatar/Workspace/sgrep-rules/go/gosec/bad_crypto/use_of_weak_crypto_md5.go
rule:go.gosec.bad_crypto.use_of_md5: Use of weak cryptographic primitive MD5
26:	h := md5.New()

/Users/ulziibayarotgonbaatar/Workspace/sgrep-rules/go/gosec/bad_crypto/use_of_weak_crypto_rc4.go
rule:go.gosec.bad_crypto.use_of_rc4: Use of weak cryptographic primitive rc4
10:	c, err := rc4.NewCipher(key)

/Users/ulziibayarotgonbaatar/Workspace/sgrep-rules/go/gosec/bad_crypto/use_of_weak_crypto_sha1.go
rule:go.gosec.bad_crypto.use_of_sha1: Use of weak cryptographic primitive SHA1
18:	h := sha1.New()
```

### CODE REVIEW

**Comment by ulziibay** [comment]
Linked to https://github.com/returntocorp/enterprise/issues/124

File: go/gosec/bad_crypto/use_of_weak_crypto.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  - id: use_of_md5
+    message:
+      Use of weak cryptographic primitive MD5
+    languages: [go]
+    severity: WARNING
+    metadata:
+      owasp: 'A9: Using Components with Known Vulnerabilities'
+      cwe: 'CWE-327: Use of a Broken or Risky Cryptographic Algorithm'
+      source_rule_url: https://github.com/securego/gosec#available-rules
+    patterns:
```
Comment (mschwager) [inline go/gosec/bad_crypto/use_of_weak_crypto.yaml:11]: `pattern-either` can now be used as a top-level key, so we don't need to nest it underneath `patterns` anymore. Same to the below usages.

Comment (ulziibay) [reply go/gosec/bad_crypto/use_of_weak_crypto.yaml:11]: ok

File: go/gosec/bad_crypto/use_of_weak_crypto.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
+rules:
+  - id: use_of_md5
```
Comment (mschwager) [inline go/gosec/bad_crypto/use_of_weak_crypto.yaml:2]: The convention I see in `semgrep-rules` is to use `-` instead of `_` in rule IDs.

Comment (ulziibay) [reply go/gosec/bad_crypto/use_of_weak_crypto.yaml]: ack

**Pull request merged**